src/Poker/Holdem/Simulate.hs

```diff
@@ -0,0 +1,155 @@
+-- |
+-- Module      : Poker.Holdem.Simulate
+-- Description : Monte Carlo simulation of a poker holdem game.
+-- Copyright   : (c) Ghais Issa, 2021
+--
+-- Uses Monte Carlo Methods to determine the probability for each player in a game winning.
+-- The simulation randomize each of the unknown cards in the game, this could include any
+-- number of players' cards as well as flop, turn and river.
+module Poker.Holdem.Simulate
+  (
+    Player(..)
+  , Game(..)
+  , simulate
+  ) where
+
+import           Data.Random (RVar, RandomSource, runRVar)
+
+import           Control.Monad.State
+import           Data.List (transpose)
+
+import           Data.Random.Source.DevRandom (DevRandom (DevRandom))
+import           Poker.Deck
+import qualified Poker.Deck as Deck (shuffle)
+import           Poker.Holdem
+import           Poker.Holdem.Evaluate (HandRank, evaluate)
+
+-- | A player can have 0, 1, 2 known cards.
+data Player = Player
+  {
+    card1 :: Maybe Card
+  , card2 :: Maybe Card
+  }
+
+-- | An abstraction that represents a poker game, with some unknowns.
+--This allows us to simulate a game from any possible state.
+data Game = Game
+  {
+    players :: [Player]      -- ^ Players.
+  , flop    :: Maybe Flop    -- ^ The flop if known. Nothing otherwise.
```

Comment:
With my above-suggested comment for board:

data SomeBoard where
    SomeBoard :: (s :: Street) -> Board s -> SomeBoard

data Game = Game { players :: [Players] , board :: SomeBoard }